File: Add Code Here/Java/bubble_sort.java
package Sorting;

import java.util.Scanner;

public class bubble_sort {
    static void bubblesort(int arr[], int n, int i, int j)
    {
        int temp;
        for(i=0;i<n-1;i++){
            for(j=0;j<n-1;j++){
                if(arr[j]>arr[j+1]){
                    temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                }
            }
        }

    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter the size of array: ");
        int n = input.nextInt();
        int [] arr = new int[n];
        int i;
        System.out.print("Enter "+ n + " element of array: ");
        for(i=0;i<n;i++)
            arr[i]= input.nextInt();

        bubblesort(arr,n,0,n-1);
        System.out.print("sorted array: ");
        for(i=0;i<n;i++)
            System.out.print(arr[i]+ " ");
    }

}

File: Add Code Here/All_Sorting/Bubble_sort.py
def bubbleSort(arr):
    n = len(arr)
    # optimize code, so if the array is already sorted, it doesn't need
    # to go through the entire process
    swapped = False
    # Traverse through all array elements
    for i in range(n - 1):
        # range(n) also works but the outer loop will
        # repeat one time more than needed.
        # Last i elements are already in place
        for j in range(0, n - i - 1):
            # traverse the array from 0 to n-i-1
            # Swap if the element found is greater
            # than the next element
            if arr[j] > arr[j + 1]:
                swapped = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

        if not swapped:
            # if we haven't needed to make a single swap, we
            # can just exit the main loop.
            return

def main():
    n = int(input("Enter the number of elements: "))
    arr = []
    for i in range(n):
        element = int(input(f"Enter element {i+1}: "))
        arr.append(element)

    bubbleSort(arr)

    print("Sorted array is:")
    for i in range(len(arr)):
        print("%d" % arr[i], end=" ")

if __name__ == "__main__":
    main()

